lib/spack/docs/workflows.rst

```diff
@@ -1198,6 +1198,23 @@ In order to build and run the image, execute:
    # image run hook: the -l will make sure /etc/profile environments are loaded
    CMD        /bin/bash -l
 
+For examples of how we use docker in development, see
```

Comment:
Just to keep this section focused, I would move this fully in the `Best Practices` section immediately below.

New sub-sections with `""""` level:
- docker in development (this reference)
- OSX, Windows (the paragraph as text instead of a note)